extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authorization/entity/BasicAuthorizerGroupMappingFull.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.security.basic.authorization.entity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class BasicAuthorizerGroupMappingFull
+{
+  private final String name;
+  private final String groupPattern;
+  private final Set<BasicAuthorizerRole> roles;
+
+  @JsonCreator
+  public BasicAuthorizerGroupMappingFull(
```

Comment:
this class looks almost duplicate to BasicAuthorizerGroupMapping, 
Can we remove duplicates ? Or are there any differences ?

Reply:
I modeled BasicAuthorizerGroupMapping and BasicAuthorizerGroupMappingFull entities similar to already existing BasicAuthorizerUser and BasicAuthorizerUserFull entities.
BasicAuthorizerGroupMapping contains a set of role names whereas BasicAuthorizerGroupMappingFull contains a set of role objects, which contains the role name and list of permission objects associated with the role name.
BasicAuthorizerResource getUser and getGroupMapping rest endpoints allows for query parameter "full" that's used to decide whether to just return the role names or the "full" role object in the response. 
Please let me know what you think.